iothub_client/src/iothub_client_core.c

```diff
@@ -1927,22 +2019,21 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_SendReportedState(IOTHUB_CLIENT_CORE_HANDL
     return result;
 }
 
-IOTHUB_CLIENT_RESULT IoTHubClientCore_GetTwinAsync(IOTHUB_CLIENT_CORE_HANDLE iotHubClientHandle, IOTHUB_CLIENT_DEVICE_TWIN_CALLBACK deviceTwinCallback, void* userContextCallback)
+static IOTHUB_CLIENT_RESULT getTwinAsync(IOTHUB_CLIENT_CORE_HANDLE iotHubClientHandle, IOTHUB_TWIN_REQUEST_OPTIONS_HANDLE twinRequestOptions,
+                                         void* twinCallback, void* userContext, DEVICE_TWIN_MESSAGE_TYPE twinMessageType)
 {
-    IOTHUB_CLIENT_RESULT result;
+   IOTHUB_CLIENT_RESULT result;
 
-    // Codes_SRS_IOTHUBCLIENT_09_009: [ If `iotHubClientHandle` or `deviceTwinCallback` are `NULL`, `IoTHubClientCore_GetTwinAsync` shall return `IOTHUB_CLIENT_INVALID_ARG`. ]
-    if (iotHubClientHandle == NULL || deviceTwinCallback == NULL)
+    // Codes_SRS_IOTHUBCLIENT_09_009: [ If `iotHubClientHandle` is `NULL`, `getTwinAsync` shall return `IOTHUB_CLIENT_INVALID_ARG`. ]
+    if (iotHubClientHandle == NULL)
     {
         result = IOTHUB_CLIENT_INVALID_ARG;
-        LogError("Invalid argument (iotHubClientHandle=%p, deviceTwinCallback=%p)", iotHubClientHandle, deviceTwinCallback);
+        LogError("Invalid argument (iotHubClientHandle=%p)", iotHubClientHandle);
     }
     else
     {
-        IOTHUB_CLIENT_CORE_INSTANCE* iotHubClientInstance = (IOTHUB_CLIENT_CORE_INSTANCE*)iotHubClientHandle;
```

Comment:
sometime this is useful while debugging.

Reply:
put back